plugin-server/src/main/ingestion-queues/session-recording/blob-ingester/session-manager.ts

```diff
@@ -163,12 +169,28 @@ export class SessionManager {
             return
         }
 
-        const bufferAge = Date.now() - this.buffer.oldestKafkaTimestamp
-        const tolerance = this.serverConfig.SESSION_RECORDING_MAX_BUFFER_AGE_SECONDS * 1000
-        const tenHoursInMilliseconds = 10 * 60 * 60 * 1000
-        const isLaggingALot = bufferAge >= tenHoursInMilliseconds
-        if (bufferAge >= tolerance) {
-            if (this.chunks.size > 0 && isLaggingALot) {
+        if (this.buffer.oldestKafkaTimestamp === null) {
+            // We have no messages yet, so we can't flush
+            if (this.buffer.count > 0) {
+                throw new Error('Session buffer has messages but oldest timestamp is null. A paradox!')
+            }
+            return
+        }
+
+        const bufferAge = referenceNow - this.buffer.oldestKafkaTimestamp
+
+        if (bufferAge >= flushThresholdMillis) {
+            const logContext = {
+                bufferAge,
+                sessionId: this.sessionId,
+                partition: this.partition,
+                chunkSize: this.chunks.size,
+                oldestKafkaTimestamp: this.buffer.oldestKafkaTimestamp,
+                referenceTime: referenceNow,
+                flushThresholdMillis,
+            }
+
+            if (this.chunks.size > 0) {
```

Comment:
We should _not_ do this btw. It could be just chance that the flush comes in between two chunks being processed...

Doesn't have to be fixed now but as a follow up we should move the chunk dropping into its own thing using some sort of timestamp comparison. i.e. `before flush -> check age of offsets and if they are too old just drop them`

Reply:
Yep, this chunk handling is poor 😓 

But, so long as we see that generally the lag handling and chunk handling fixes the backfill problem then I'll move onto chunk handling properly as the next job